docs/how-to/add-custom-code.md

```diff
@@ -1,95 +1,78 @@
-
-### The 'entity' folder
-
-Each entity has a dedicated folder under the 'src' folder.
-For example:
+Each entity has a dedicated folder under the 'src' folder, containing all necessary modules and files:
 
 ```
 └── src
-	├── customer
-	├── user
-	├── project
-	└── task
+    ├── customer
+    ├── user
+    ├── project
+    └── task
 ```
 
-The entity folder contains all the modules and files required to work with a specific entity, i.e. GraphQL resolver, REST API controller, service, and DTOs.
+Within each entity folder, files are split into two groups:
 
-The files are split into two groups:
+1. **Base files**: Located in the 'base' folder, these are automatically generated by Amplication with every change. **They should not be altered as they will be overwritten with each new build**.
 
-- Base files - the base files are the ones that are automatically generated by Amplication with every change. These files should not be altered or customized as they will be overwritten and the changes will not be preserved. All the base files exist in a separate folder named "base".
-
-- Customizable files - the customizable files inherit from the base files and they can be customized and will never be overwritten by Amplication. All the customizable files reside directly in the entity folder and can be freely customized and altered.
+2. **Non-base Customizable files**: These inherit from the base files and can be safely customized with your custom code. They reside directly in the entity folder.
 
 ```
 src
 └── Customer
     ├── Base
-    │    ├── CreateCustomerArgs.ts
-    │    ├── CustomerFindManyArgs.ts
-    │    ├── CustomerFindUniqueArgs.ts
-    │    ├── customer.controller.base.spec.ts
     │    ├── customer.controller.base.ts
-    │    ├── customer.resolver.base.ts
     │    ├── customer.service.base.ts
-    │    ├── customer.ts
-    │    ├── CustomerCreateInput.ts
-    │    ├── CustomerUpdateInput.ts
-    │    ├── CustomerWhereInput.ts
-    │    ├── CustomerWhereUniqueInput.ts
-    │    ├── DeleteCustomerArgs.ts
-    │    └── updateCustomerArgs.ts
-    ├── customer.controller.spec.ts
+    │    └── ...
     ├── customer.controller.ts
-    ├── customer.module.ts
-    ├── customer.resolver.ts
-    └── customer.service.ts
-
+    ├── customer.service.ts
+    └── ...
 ```
 
-### The 'base' folder
+Amplication may still update the non-base files that include your custom code in certain circumstances. These updates are limited to changes necessary for preventing errors and ensuring the project can build correctly. For example, if you remove a plugin that was previously used in your service, Amplication might update the non-base files to remove references to the removed plugin, thus preventing build errors.
+
+This approach allows Amplication to maintain the integrity of your project structure while still preserving your custom code to the greatest extent possible.
```

Comment:
I think we need to emphasize here that Amplication code will always respect the user's custom code, and won't overwrite it.